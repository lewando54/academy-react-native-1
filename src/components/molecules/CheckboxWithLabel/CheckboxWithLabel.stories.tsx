/* eslint-disable @typescript-eslint/consistent-type-assertions */
import { StoryObj, Meta } from '@storybook/react'
import { useState } from 'react'

import CheckboxWithLabel from './CheckboxWithLabel'
import { create } from 'react-test-renderer'

interface IMockupCheckboxProps {
    bgColor?: string 
    checkmarkColor?: string
    labelText: string
}

function StatefulCheckboxWithLabel({bgColor, checkmarkColor, labelText}: IMockupCheckboxProps) {
    let [checked, setChecked] = useState(false)

    return (
        <CheckboxWithLabel labelText={labelText} checked={checked} testId='test' bgColor={bgColor} checkmarkColor={checkmarkColor} onChange={(newValue) => {setChecked(newValue)}}/>
    )
}

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
  component: StatefulCheckboxWithLabel,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
    bgColor: { control: 'color' },
    checkmarkColor: { control: 'color' },
    labelText: { control: 'text' }
  }
} satisfies Meta<typeof StatefulCheckboxWithLabel>

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
export default meta
type Story = StoryObj<typeof meta>

export const Primary: Story = {
    args: {
        bgColor: 'rgb(236, 72, 153)',
        checkmarkColor: '#fff',
        labelText: 'Check me'
    }
}
// More on args: https://storybook.js.org/docs/react/writing-stories/args
