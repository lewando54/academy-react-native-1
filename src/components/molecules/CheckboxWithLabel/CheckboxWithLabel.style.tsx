import { StyleSheet } from 'react-native'

export default StyleSheet.create({
    div: {
        display: 'flex',
        width: '100%',
        flexDirection: 'row',
        gap: 8,
        alignItems: 'center'
    },
    label: {
        fontSize: 16
    }
})