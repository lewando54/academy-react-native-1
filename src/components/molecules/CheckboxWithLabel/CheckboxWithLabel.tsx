import React, { forwardRef } from 'react'
import Checkbox, { ICheckboxProps, ICheckboxRef } from '../../atoms/Checkbox/Checkbox'
import { View, Text } from 'react-native'
import checkboxWithLabelStyle from './CheckboxWithLabel.style'

export interface ICheckboxWithLabelProps extends ICheckboxProps{
    labelText: string
}

function CheckboxWithLabel({labelText, ...props}: ICheckboxWithLabelProps, ref: React.Ref<ICheckboxRef>){
    return(
        <View style={checkboxWithLabelStyle.div}>
            <Checkbox {...props} ref={ref}/>
            <Text style={checkboxWithLabelStyle.label}>{labelText}</Text>
        </View>
    )
}

export default forwardRef(CheckboxWithLabel)