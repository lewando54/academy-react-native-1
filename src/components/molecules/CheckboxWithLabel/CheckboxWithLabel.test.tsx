import React from 'react'
import { render } from '@testing-library/react-native'
import * as renderer from 'react-test-renderer'
import CheckboxWithLabel from './CheckboxWithLabel'

describe('CheckboxWithLabel', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<CheckboxWithLabel labelText='Testing' onChange={() => {}} checked={false} testId='test' />)
    expect(tree).toMatchSnapshot()
  })

  it('should render label with text', () => {
    const { getByText } = render(<CheckboxWithLabel labelText='Testing' onChange={() => {}} checked={false} testId='test' />)
    expect(getByText('Testing')).toBeTruthy()
  })
})