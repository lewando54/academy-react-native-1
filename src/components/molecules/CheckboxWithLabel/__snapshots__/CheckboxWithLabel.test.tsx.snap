// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxWithLabel should render correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "gap": 8,
      "width": "100%",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": false,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "height": 25,
          "position": "relative",
          "width": 25,
        },
        undefined,
      ]
    }
    testID="test"
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "transparent",
          "borderColor": "#ddd",
          "borderRadius": 4,
          "borderWidth": 2,
          "height": 25,
          "left": 0,
          "opacity": 1,
          "position": "absolute",
          "top": 0,
          "width": 25,
        }
      }
    />
    <View
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgb(236, 72, 153)",
          "borderColor": "rgb(236, 72, 153)",
          "borderRadius": 4,
          "borderWidth": 2,
          "display": "flex",
          "height": 25,
          "justifyContent": "center",
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "top": 0,
          "width": 25,
        }
      }
      testID="checkbox-view"
    >
      <View
        collapsable={false}
        style={
          {
            "transform": [
              {
                "rotate": "180deg",
              },
              {
                "scale": 0,
              },
            ],
          }
        }
      >
        <Text />
      </View>
    </View>
  </View>
  <Text
    style={
      {
        "fontSize": 16,
      }
    }
  >
    Testing
  </Text>
</View>
`;
